### PR TITLE
adding firestore emulator client

### PR DESCRIPTION
**Pull request by codestiff**
This works but the code isn't cleaned up.
Looking for an initial review

### CODE REVIEW

File: src/clj/firestore_clj/google.clj [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  (let [credentials (mk-fake-credentials)
+        channel-provider (mk-emulator-grpc-channel-provider emulator-host)
+        ^FirestoreOptions options (->
+                                   (FirestoreOptions/getDefaultInstance)
+                                   (.toBuilder)
+                                   ^FirestoreOptions$Builder
+                                   (.setProjectId project-id)
+                                   (.setChannelProvider channel-provider)
+                                   ^FirestoreOptions$Builder
+                                   (.setCredentials credentials)
```
Comment (codestiff) [inline src/clj/firestore_clj/google.clj:57]: As far as I can tell the fake credentials are not used, except to guarantee validation works for the builder.

The real authorization happens using the `fixed-header-provider`

https://github.com/polvotech/firestore-clj/pull/4/files#diff-8dcfb082370a13ada9fe204cec4f7a1aR27

File: project.clj [hunk 2]
```diff
@@ -9,8 +9,19 @@
                  [org.clojure/core.match "1.0.0"]
                  [com.google.firebase/firebase-admin "6.12.2"]
                  [manifold "0.1.8"]]
+  :aot [
+        firestore-clj.google.fake-credentials
+        firestore-clj.google.emulator-channel-configurator
+        ]
+  :profiles {:dev {:dependencies [[com.taoensso/timbre "4.10.0"]
+                                  [com.fzakaria/slf4j-timbre "0.3.19"]
+                                  [com.google.cloud/google-cloud-logging-logback "0.116.0-alpha"]]
```
Comment (codestiff) [inline project.clj:18]: Used to pick up `grpc` logs in order to debug why auth was failing.

File: src/clj/firestore_clj/google.clj [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                                   ^FirestoreOptions (.build))]
+    (.getService options)))
+
+(comment
+  (require '[taoensso.timbre :as log])
+  (require '[firestore-clj.core :as f])
+  (def my-db (emulator-client "test-app-4" "localhost:8080"))
```
Comment (codestiff) [inline src/clj/firestore_clj/google.clj:64]: Example usage:
```
bash$ firebase emulators:start --only firestore
⚠  Could not find config (firebase.json) so using defaults.
i  emulators: Starting emulators: firestore
⚠  firestore: Did not find a Cloud Firestore rules file specified in a firebase.json config file.
⚠  firestore: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration.
i  firestore: firestore emulator logging to firestore-debug.log
✔  firestore: firestore emulator started at http://localhost:8080
i  firestore: For testing set FIRESTORE_EMULATOR_HOST=localhost:8080
✔  emulators: All emulators started, it is now safe to connect.
```

```clojure
(emulator-client "test-app-4" "localhost:8080") 
  => Firestore instance.
     Collections: ["accounts"]
```

**Comment by lurodrigo** [comment]
Thank for you for the great contribution! I'm quite busy for now and tomorrow, but I'll surely review and hopefully merge by the weekend!

**Comment by lurodrigo** [comment]
Merged! Didn't make any significant editing, just moved `emulator-client` to `firestore-clj.core` so to make it easier to refer to. Released in `1.2.1` as experimental.

If you could also add comments describing in more detail what is happening, since it seems we're being hacky here, it would be great for future maintenance. Thank you again!

**Comment by lurodrigo** [comment]
I also took the liberty of adding you to the contributors section of the README, but can delete that if it's your wish.

**Comment by codestiff** [comment]
> If you could also add comments describing in more detail what is happening, since it seems we're being hacky here, it would be great for future maintenance. Thank you again!

Sure thing. And yes it is being hacky. Although it is not explicitly stated in the java source code. It looks like the client is being "defensive" and only wants to allow you to configure the emulator client if you set the environment variable `FIRESTORE_EMULATOR_HOST`.

**Pull request merged**